polynote-kernel/src/main/scala/polynote/kernel/interpreter/python/PythonInterpreter.scala

```diff
@@ -255,29 +255,29 @@ class PythonInterpreter private[python] (
 
   // TODO: we probably need to write out prior code cells, and use the jedi Project (https://jedi.readthedocs.io/en/latest/docs/api.html#jedi.Project)
   //        to allow finding references to other cells. This will only find references that are external.
-  override def goToDefinition(code: String, pos: Int, state: State): RIO[Blocking with Logging, (List[DefinitionLocation], Option[String])] = {
+  override def goToDefinition(code: String, pos: Int, state: State): RIO[Blocking with Logging, List[DefinitionLocation]] = {
     for {
       globals     <- populateGlobals(state)
       (line, col)  = posToLineAndColumn(code, pos)
       jedi        <- jep(_.getValue("jedi.Interpreter", classOf[PyCallable]).callAs(classOf[PyObject], code, Array(globals)))
       locations   <- definitionLocations(jedi, line, col)
-      source      <- definitionSource(locations)
-    } yield (locations, source)
-  }.catchAllCause(cause => Logging.error(cause).as((Nil, None)))
+    } yield locations
+  }.catchAllCause(cause => Logging.error(cause).as(Nil))
 
-  override def goToDependencyDefinition(uri: String, pos: Int): RIO[Blocking, (List[DefinitionLocation], Option[String])] = for {
+  override def goToDependencyDefinition(uri: String, pos: Int): RIO[Blocking, List[DefinitionLocation]] = for {
     path        <- effect(new URI(uri).getPath)
     // need this just for line/column. Probably should have just used that for position so it could come from front-end
     codeStr     <- readFile(Paths.get(path))
     (line, col)  = posToLineAndColumn(codeStr, pos)
     script      <- jep(_.getValue("jedi.Script", classOf[PyCallable]).callAs(classOf[PyObject], Map[String, Object]("path" -> path).asJava))
     locations   <- definitionLocations(script, line, col)
-    source      <- definitionSource(locations)
-  } yield (locations, source)
+  } yield locations
 
-  override def getDependencyContent(uri: String): RIO[Blocking, String] = ZIO {
-    val parsed = new URI(uri)
-    Files.readAllLines(Paths.get(parsed.getPath)).asScala.mkString("\n")
+  override def getDependencyContent(uri: String): RIO[Blocking, String] = ZIO(new URI(uri)).flatMap {
+    // TODO: some better checking that it's OK to read this path. Maybe goToDefinition should keep track of
```

Comment:
+1 to this idea. As it is now, you can hit `/dependency/path/to/anything` and see any file on the machine, right? Could you even do ``/dependency/../../../path/to/anything`?

Reply:
Well, it has to be a .py file. But yes. The urls would be absolute, too — no need for the ..s

Reply:
We can disable this for now, since locating the definitions doesn't seem to work anymore anyway.

Reply:
I suppose it would. Anyway, I'll just turn this method into a stub before merging (since the relevant jedi functionality seems to have stopped working anyhow). When we are able to properly support python jump-to-definition, we can implement the idea in the TODO.